### ISSUE TITLE
vagrant up: [ErrorException] my-project: rename(/home/vagrant/.composer/cache/composer-temp.phar,/usr/local/bin/composer): Permission denied

### ISSUE THREAD

**Issue by rvzug** [laravel/homestead#1051]
## Versions
  - Vagrant: 2.2.3
  - Provider: Virtualbox (5.2.26)
  - Homestead: 8.0.1 | Box version: v7.0.0

## Host operating system
Windows 10 64-bits

## Homestead.yaml
```yaml
ip: 192.168.10.10
memory: 2048
cpus: 1
provider: virtualbox
authorize: ~/.ssh/id_rsa.pub
keys:
    - ~/.ssh/id_rsa
folders:
    -
        map: 'D:\Projects\my-prject'
        to: /home/vagrant/code
sites:
    -
        map: my-project.test
        to: /home/vagrant/code/public
databases:
    - homestead
name: my-project
hostname: my-project
```

## Vagrant destroy & up output
[GitHub Gist](https://gist.github.com/rvzug/a1981911b6aef70fbac6b616c355db51) 

## Expected behavior
Update composer trough self-update and continue until ``vagrant up`` completes

## Actual behavior
[ErrorException] my-project: rename(/home/vagrant/.composer/cache/composer-temp.phar,/usr/local/bin/composer): Permission denied

## Steps to reproduce
1. Vanilla installation of Homestead 
2. ``vagrant up``

## References
#1049

**Comment by svpernova09** [comment]
Verified this is an issue. I'm not sure why it's not happening on the other providers. Will require a base box update to fix it.

Until then you can resolve this my running `sudo chown -R vagrant:vagrant /usr/local/bin && sudo composer self-update`

Even those this looks like a hard error the environment is mostly find. Anything in `after.sh` hasn't been run.

**Comment by rvzug** [comment]
Thanks. The issue #1049 describes the problem using the hyper-v provider... Also I saw that Composer released the new 1.8.2 version today.

Well, the **dirty**-hack I've used to cut around the issue:
1) Use 'project based' installation (https://laravel.com/docs/5.7/homestead#per-project-installation)
2) comment line 486-490 in vendor/laravel/homestead/scripts/homestead.rb, lines will be:
```rb
    #config.vm.provision 'shell' do |s|
    #  s.name = 'Update Composer'
    #  s.inline = 'sudo -u vagrant /usr/local/bin/composer self-update --no-progress && sudo chown -R vagrant:vagrant /home/vagrant/.composer/'
    # s.privileged = false
    #end
```
3) ``vagrant up`` no issue on the composer self-update anymore and continues normally
4) ``vagrant ssh`` and ``sudo chown -R vagrant:vagrant /usr/local/bin && sudo composer self-update`` as @svpernova09 described.
5) Up and running normally as expected.

**Comment by svpernova09** [comment]
Resolved via #1052  and in release v8.0.2
